### ISSUE TITLE
libzerocash's getRandomBytes gives no indication of failure

### ISSUE THREAD

**Issue by defuse** [zcash/zcash#450]
```
void getRandBytes(unsigned char* bytes, int num) {
    int ret = RAND_bytes(bytes, num);
    if(ret != 1)
        std::cout << "rand_bytes error!" << ERR_get_error() << std::endl;
}
```

**Comment by paragonie-scott** [comment]
Related to #515 (is one a duplicate of the other? either way, using libsodium's CSPRNG would fix both).

**Comment by daira** [comment]
Thanks @paragonie-scott , I consider this a serious security bug.

**Comment by ebfull** [comment]
This was fixed by #793.
